### PR TITLE
NISP-2067: Future Proof Personal Max message

### PR DESCRIPTION
**Pull request by dspork**
* This PR future proofs the personal max information on the account page.
* As all years are not worth the same and may not add to their pension, we can no longer say "fill any year", "you only need to fill X years". So we have removed this information
* This does not apply if you literally only have one gap in your NI Record, so they will get the regular messaging
* I have added a feature switch as we don't need this to 2017
* As this is a feature switch, I have wrote the unit tests for both situations | @swaistle, @InduTest could you look at these and see if they are enough that we don't need acceptance tests?

### CODE REVIEW

File: app/uk/gov/hmrc/nisp/controllers/AccountController.scala [hunk 1]
```diff
@@ -91,8 +91,15 @@ trait AccountController extends NispFrontendController with AuthorisedForNisp wi
...
-            Ok(account(nino, spSummary, currentChart, forecastChart, personalMaximumChart, isPertax))
-               .withSession(storeUserInfoInSession(user, spSummary.contractedOutFlag))
+            Ok(account(
+              nino,
+              spSummary,
+              currentChart,
+              forecastChart,
+              personalMaximumChart,
+              isPertax,
+              futureProofPersonalMax = applicationConfig.futureProofPersonalMax && spSummary.numberOfGaps > 1
```
Comment (andrewgee) [inline app/uk/gov/hmrc/nisp/controllers/AccountController.scala]: It seems strange that you're combining the config property and number of gaps rule into something named the same as the config property?

Comment (dspork) [inline app/uk/gov/hmrc/nisp/controllers/AccountController.scala]: What would you suggest as an alternate name?

**Comment by andrewgee** [comment]
:+1: :ship: :question: :hamster:

**Comment by InduTest** [comment]
@dspork I can see that the message for single year is been changed a bit when compared to the current one. If this the expected message when config is off, it looks good for me. And everything else looks good. We can however do manual exploratory testing once ready.

**Pull request merged**